Apps/Sandcastle/CesiumSandcastle.js

```diff
@@ -821,6 +822,28 @@ require({
         }
     }
 
+    registry.byId('buttonNew').on('click', function() {
+        var htmlText = (htmlEditor.getValue()).replace(/\s/g, '');
+        var jsText = (jsEditor.getValue()).replace(/\s/g, '');
+        var confirmChange = true;
+        if (demoHtml !== htmlText || demoJs !== jsText) {
+            confirmChange = window.confirm('You have unsaved changes. Are you sure you want to navigate away from this demo?');
+        }
+        if(confirmChange){
+            loadFromGallery(newDemo);
+            var demoSrc = newDemo.name + '.html';
+            var queries = window.location.search.substring(1).split('&');
+            for(var i = 0; i < queries.length; i++){
+                var key = queries.split('=')[0];
+                if(key === "src"){
+                    if (demoSrc !== queries.split('=')[1]) {
```

Comment:
Actually this is on 837 and 839.  Maybe `queries[i].split('=')` should be its own variable inside the loop.